Time sync

patnolan33 opened the pull request.

kkroeger007 commented:
There were some changes committed to further generalize the function names and give the end-user some more flexibility in understanding. However, I would recommend using the event API associated with each of the modules rather than publishing a topic and explicitly listening for the string name of the topic and its accompanying variables. Gives a bad point of failure if any of those definitions change in the future. Would recommend readdressing this issue.

Additionally, as implemented there should exist a check that evaluates the GPS fix before publishing the clock update times. The clock on the autopilot that evaluates this condition only fuses information based on what is known, w/o a proper fix this could be a corrupted measurement. Ideally the check would consist of both has data changed && fixType > 3D.

patnolan33 commented:
Moved system time update away from the explicit check for a string topic name and instead moved to an event. the `GVEvents_NewSystemTime` event should trigger a new update to the core's delta time based on the system clock. 

Also added a check for GPS status and that the data is actually different prior to sending data to the core (see changes to `src/module_vehicle_MAVLINK/vehicle_object/parse_mavlink.cpp` in da1002b7ddff3416f2169ac568e90944bc788852)
